compiler/rustc_passes/src/entry.rs

```diff
@@ -85,7 +60,11 @@ fn attr_span_by_symbol(ctxt: &EntryContext<'_>, id: ItemId, sym: Symbol) -> Opti
 fn find_item(id: ItemId, ctxt: &mut EntryContext<'_>) {
     let at_root = ctxt.tcx.opt_local_parent(id.owner_id.def_id) == Some(CRATE_DEF_ID);
 
-    match entry_point_type(ctxt, id, at_root) {
+    let attrs = ctxt.tcx.hir().attrs(id.hir_id());
+    let entry_point_type = rustc_ast::entry::entry_point_type(attrs, at_root, || {
+        ctxt.tcx.opt_item_name(id.owner_id.to_def_id())
```

Comment:
`get_name` does not really need to be a closure. Passing an `Option<Symbol>` directly is sufficient. `opt_item_name` only looks at def-path information, and does not involve any query. It should be cheap enough to call unconditionally.

Reply:
Ok.